src/TLuaInterpreter.cpp

```diff
@@ -17923,3 +17925,75 @@ int TLuaInterpreter::setMapRoomExitsColor(lua_State* L)
     lua_pushboolean(L, true);
     return 1;
 }
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#prompt
+int TLuaInterpreter::prompt(lua_State *L)
+{
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "prompt: bad argument #1 type (dialog title as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    if (!lua_isstring(L, 2)) {
+        lua_pushfstring(L, "prompt: bad argument #2 type (dialog label as string expected, got %s!)", luaL_typename(L, 2));
+        return lua_error(L);
+    }
+
+    if (!lua_isfunction(L, 3)) {
+        lua_pushfstring(L, "prompt: bad argument #3 type (callback as function expected, got %s!)", luaL_typename(L, 3));
+        return lua_error(L);
+    }
+
+    if (lua_gettop(L) >= 4 && !lua_istable(L, 4)) {
+        lua_pushfstring(L, "prompt: bad argument #4 type (option list as table expected, got %s!)", luaL_typename(L, 4));
+        return lua_error(L);
+    }
+
+    auto *dialog = new QInputDialog(mudlet::self(), Qt::Dialog);
+    dialog->setWindowTitle(lua_tostring(L, 1));
+    dialog->setLabelText(lua_tostring(L, 2));
+    dialog->setOkButtonText(tr("OK"));
```

Comment:
Is this and the next necessary or do they get populated automagically by the selection of the Qt translation by the GUI language setup code?

Reply:
I was pretty sure, that this will work for translation, but I guess I have to check :)

Reply:
I guess we can extend and add table with options, sure there is more to QInputDialog then exposed here. Actually while creating example for @vadi2 I realized terrible mistake I've made. There is really good reason to handle cancel clicks as well.

Reply:
Actually `tr` does not work on translations I guess :/ We can delete both lines and we'll get standard OK and Cancel

Reply:
>  Would it be a better UI experience if we could provide more access to the `QInputDialog` class - I note that there is scope for asking for a freeform string (possibly with a predefined value, and various echo modes including a password one that only shows the characters as they are entered but otherwise masked), an integer (with a predefined value and min / max settings) ....
> 
> ... perhaps it would be a more usable API if it took various "named" parameters from a table of options with one of them determining how the rest are to be used. There are a lot of potentially useful features in this Qt class and it might be possible to start off with something simple and develop it over time! For instance it could be set as read-only and pop up a value for the user to _copy_ from for their own use...

Maybe instead of creating type prompt and handling different option sets would be better to create prompt family functions, so like `promptForNumber`, `promptForPassword` and here not to include dozens of parameters it could have option table specific for flavour. But this is going a bit deeper into details, password dialog actually has a chance to be least used api of all time :) And `promptForNumber` while useful, can be added independently.

Additionally read-only popup is not that bad thing actually, but again - could be different API function, since it does not require accept function callback etc. I guess QT has some clipboard access capabilities, so it could be made even easier.